components/scream/cmake/ScreamUtils.cmake

```diff
@@ -44,90 +44,57 @@ function(CreateUnitTest target_name target_srcs scream_libs)
   cmake_parse_arguments(PARSE_ARGV 3 CreateUnitTest "${options}" "${oneValueArgs}" "${multiValueArgs}")
   CheckMacroArgs(CreateUnitTest CreateUnitTest "${options}" "${oneValueArgs}" "${multiValueArgs}")
 
-  set (TEST_INCLUDE_DIRS
-       ${SCREAM_INCLUDE_DIRS}
-       ${CMAKE_CURRENT_SOURCE_DIR}
-       ${CMAKE_CURRENT_BINARY_DIR}
-       ${SCREAM_F90_MODULES}
-       ${CreateUnitTest_INCLUDE_DIRS}
+  set(TEST_INCLUDE_DIRS
+    ${SCREAM_INCLUDE_DIRS}
+    ${CMAKE_CURRENT_SOURCE_DIR}
+    ${CMAKE_CURRENT_BINARY_DIR}
+    ${SCREAM_F90_MODULES}
+    ${CreateUnitTest_INCLUDE_DIRS}
   )
 
-  set (test_libs "${scream_libs}")
+  set(test_libs "${scream_libs}")
   list(APPEND test_libs "${SCREAM_TPL_LIBRARIES}")
 
-  # Note: some args are likely empty, so check them before adding them,
-  #       or you would have either a) a keyword followed by nothing (error),
-  #       or b) to add a space at front or back (causing EKAT to think that
-  #       that option is present, when it's not).
-  set (cut_options EXCLUDE_TEST_SESSION)
-  if (CreateUnitTest_EXCLUDE_MAIN_CPP)
-    list(APPEND cut_options EXCLUDE_MAIN_CPP)
-  endif()
-  if (CreateUnitTest_SERIAL)
-    list(APPEND cut_options SERIAL)
-  endif()
-  if (CreateUnitTest_PRINT_OMP_AFFINITY)
-    list(APPEND cut_options PRINT_OMP_AFFINITY)
-  endif()
-  if (CreateUnitTest_DEP)
-    list(APPEND cut_options DEP ${CreateUnitTest_DEP})
-  endif()
-  if (CreateUnitTest_PROPERTIES)
-    list(APPEND cut_options PROPERTIES ${CreateUnitTest_PROPERTIES})
-  endif()
-  if (CreateUnitTest_LABELS)
-    list(APPEND cut_options LABELS ${CreateUnitTest_LABELS})
-  endif()
-  if (CreateUnitTest_MPI_RANKS)
-    list(APPEND cut_options MPI_RANKS ${CreateUnitTest_MPI_RANKS})
-  endif ()
-  if (CreateUnitTest_THREADS)
-    list(APPEND cut_options THREADS ${CreateUnitTest_THREADS})
-  endif ()
-  if (CreateUnitTest_EXE_ARGS)
-    list(APPEND cut_options EXE_ARGS ${CreateUnitTest_EXE_ARGS})
-  endif ()
+  #
+  # Convert CreateUnitTest options/args to EkatCreateUnitTest options/args
+  #
+  set(skip_list "INCLUDE_DIRS;COMPILER_F_FLAGS")
+
+  set(cut_options EXCLUDE_TEST_SESSION)
+
+  foreach(item IN LISTS options oneValueArgs multiValueArgs)
+    if (NOT item IN_LIST skip_list AND CreateUnitTest_${item})
+      if (item IN_LIST options)
+        list(APPEND cut_options ${item})
+      else()
+        list(APPEND cut_options ${item} ${CreateUnitTest_${item}})
+      endif()
+    endif()
+  endforeach()
+
   if (SCREAM_MPI_EXTRA_ARGS)
     list(APPEND cut_options MPI_EXTRA_ARGS ${SCREAM_MPI_EXTRA_ARGS})
   endif ()
-  if (CreateUnitTest_COMPILER_DEFS)
-    list(APPEND cut_options COMPILER_DEFS ${CreateUnitTest_COMPILER_DEFS})
-  endif ()
-  if (CreateUnitTest_COMPILER_C_DEFS)
-    list(APPEND cut_options COMPILER_C_DEFS ${CreateUnitTest_COMPILER_C_DEFS})
```

Comment:
I noticed a lot of these hardcoded appends were not appending the same variable in the if check. For example, `CreateUnitTest_C_COMPILER_DEFS` was appended in multiple cases. I assume this was a copy/paste mistake and not something intentional?

Reply:
Yeah, this looks a lot like a copy+paste error...